controllers/mpi/mpijob_controller.go

```diff
@@ -216,6 +217,20 @@ func (r *MPIJobReconciler) SetClusterSpec(ctx context.Context, job interface{},
 	if !ok {
 		return fmt.Errorf("%+v is not a type of MPIJob", job)
 	}
+	workerReplicas := int32(0)
+	if workerSpec := mpiJob.Spec.MPIReplicaSpecs[training.MPIReplicaTypeWorker]; workerSpec != nil && workerSpec.Replicas != nil {
+		workerReplicas = *workerSpec.Replicas
+	}
+	if launcherSpec, ok := mpiJob.Spec.MPIReplicaSpecs[training.MPIReplicaTypeLauncher]; ok && launcherSpec != nil {
+		// MPIJob ensures job-attached configuration is kept by ConfigMap when launcher
+		// neither succeed nor failed.
+		isGPULauncher := resource_utils.PodRequestsForGPU(&corev1.Pod{Spec: launcherSpec.Template.Spec}) && launcherRunsWorkload
```

Comment:
As we've discussed(https://github.com/kubedl-io/kubedl/pull/129), directly use the launcherRunsWorkload variable is better?

Reply:
done